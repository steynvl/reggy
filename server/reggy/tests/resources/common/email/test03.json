{
  "type": "Email address",
  "information": {
    "username": "Specific user names only",
    "domainName": "Allow any domain on specific TLD(s)",
    "mailtoPrefix": "Optional prefix",
    "specificUserNamesOnly": "steyn;jake",
    "domainOnSpecificTld": "com;org",
    "anySubDomainOnSpecificDomain": "",
    "specificDomainsOnly": ""
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}
