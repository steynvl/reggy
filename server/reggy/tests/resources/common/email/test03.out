Java
Pattern regex = Pattern.compile("(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)");
(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)
Perl
my $regex = /(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]++\.)+(?:com|org)/;
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]++\.)+(?:com|org)
POSIX
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]++\.)+(?:com|org)
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]++\.)+(?:com|org)
Python
regex = re.compile(r'(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)')
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)
JavaScript
const regex = /(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)/;
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)
PHP
$regex = '/(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]++\.)+(?:com|org)/';
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]++\.)+(?:com|org)
Golang
regex, _ := regexp.Compile("(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)")
(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)
Rust
let re = Regex::new(r"(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)").unwrap();
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)
C#
var re = new Regex(@"(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)");
(?:mailto:)?+(?:steyn|jake)@(?:[\da-z-]+\.)+(?:com|org)
Scala
val re = "(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)".r
(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)
Kotlin
val regex = Regex("(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)")
(?:mailto:)?+(?:steyn|jake)@(?:[\\da-z-]++\\.)+(?:com|org)